res/qml/Deck/Loop.qml

```diff
@@ -0,0 +1,294 @@
+import ".." as Skin
+import Mixxx 1.0 as Mixxx
+import Mixxx.Controls 1.0 as MixxxControls
+import Qt5Compat.GraphicalEffects
+import QtQuick 2.12
+import QtQuick.Layouts
+import QtQuick.Shapes
+import QtQuick.Controls 2.12
+import "../Theme"
+
+Rectangle {
+    id: root
+
+    required property string group
+
+    color: '#626262'
+
+    Label {
+        anchors.top: parent.top
+        anchors.horizontalCenter: parent.horizontalCenter
+        text: "Loop"
+        color: '#3F3F3F'
+    }
+
+    Mixxx.ControlProxy {
+        id: loopEnabled
+
+        group: root.group
+        key: "loop_enabled"
+    }
+
+    Mixxx.ControlProxy {
+        id: beatloopSize
+
+        group: root.group
+        key: "beatloop_size"
+    }
+
+    Mixxx.ControlProxy {
+        id: beatloopActivate
+
+        group: root.group
+        key: "beatloop_activate"
+    }
+
+    Mixxx.ControlProxy {
+        id: loopHalve
+
+        function trigger() {
+            this.value = 1;
+            this.value = 0;
+        }
```

Comment:
This looks like a workaround. Isn't this a generic functionality, that should be implemented in Mixxx.ControlProxy?